src/omv/modules/py_tf.c

```diff
@@ -803,31 +708,384 @@ STATIC void py_tf_regression_input_callback(void *callback_data,
+
     fb_alloc_mark();
     py_tf_alloc_putchar_buffer();
 
-    py_tf_model_obj_t *model = py_tf_load_alloc(model_obj);
+    py_tf_model_obj_t *model = py_tf_load_alloc(pos_args[0]);
     uint8_t *tensor_arena = fb_alloc(model->params.tensor_arena_size, FB_ALLOC_PREFER_SPEED | FB_ALLOC_CACHE_ALIGN);
 
-    mp_obj_t py_tf_classify_output_callback_data;
+    mp_obj_t output_callback_data;
+    int invoke_result;
 
-    if (libtf_invoke(model->model_data,
-                     tensor_arena,
-                     &model->params,
-                     py_tf_regression_input_callback,
-                     &array_obj,
-                     py_tf_classify_output_callback,
-                     &py_tf_classify_output_callback_data) != 0) {
+    if (MP_OBJ_IS_TYPE(pos_args[1], &mp_type_tuple) || MP_OBJ_IS_TYPE(pos_args[1], &mp_type_list)) {
+        invoke_result = libtf_invoke(model->model_data,
+                                     tensor_arena,
+                                     &model->params,
+                                     py_tf_regression_input_callback,
+                                     (void *) &pos_args[1],
+                                     py_tf_classify_output_callback,
+                                     &output_callback_data);
+    } else {
+        image_t *image = py_helper_arg_to_image(pos_args[1], ARG_IMAGE_ANY);
+        rectangle_t roi = py_helper_arg_to_roi(args[ARG_roi].u_obj, image);
+
+        py_tf_input_callback_data_t py_tf_input_callback_data;
+        py_tf_input_callback_data.img = image;
+        py_tf_input_callback_data.roi = &roi;
+        py_tf_input_callback_data.input_scale = args[ARG_input_scale].u_int;
+
+        if (args[ARG_callback].u_obj != mp_const_none) {
+            py_tf_invoke_callback_data_t py_tf_invoke_output_callback_data;
+            py_tf_invoke_output_callback_data.roi = &roi;
+            py_tf_invoke_output_callback_data.callback = args[ARG_callback].u_obj;
+            py_tf_invoke_output_callback_data.out = &output_callback_data;
+            invoke_result = libtf_invoke(model->model_data,
+                                         tensor_arena,
+                                         &model->params,
+                                         py_tf_input_callback,
+                                         &py_tf_input_callback_data,
+                                         py_tf_invoke_output_callback,
+                                         &py_tf_invoke_output_callback_data);
+        } else {
+            invoke_result = libtf_invoke(model->model_data,
+                                         tensor_arena,
+                                         &model->params,
+                                         py_tf_input_callback,
+                                         &py_tf_input_callback_data,
+                                         py_tf_classify_output_callback,
```

Comment:
```suggestion
                                         py_tf_classify_output_callback,
```
I don't think should be called `py_tf_classify_output_callback` anymore, maybe just `output_callback`.

Reply:
Sure, I have no ideas for what to call this. If you have a better name let me know.

Reply:
Done